### PR TITLE
feat: Add Windows Subsystem for Linux (WSL) support with smart network binding

### PR DESCRIPTION
**Pull request by lauritk**
### Summary

This PR adds comprehensive Windows Subsystem for Linux (WSL) support to godot-mcp, enabling seamless connectivity between a WSL-hosted MCP server and Windows-hosted Godot editor. The implementation includes automatic WSL detection, smart network connection strategy, and enhanced UI controls for network configuration.

#### 🔧 WSL Support
- **Auto-detection**: Automatically detects WSL environment via environment variables and version
- **Smart connection**: Auto-discovers Windows host IP from WSL to connect to Godot running on Windows
- **Host IP discovery**: Resolves Windows host IP from WSL via `ip route`
- **Environment overrides**: `GODOT_HOST` and `GODOT_PORT` environment variables for manual configuration

#### 🎨 Enhanced Godot Plugin UI
- New **MCP status panel** in Godot editor with connection diagnostics
- **Bind mode selector**: Localhost / WSL / Custom binding options
- **Connection details**: Displays remote host:port of connected clients
- **Port override**: Configure custom listen port with UI toggle
- Real-time status updates with WSL indicators

#### 🧪 Comprehensive Testing
- Full test coverage for WSL detection across different environments
- Host IP resolution tests for various WSL scenarios
- Connection strategy selection logic validation
- Edge case and error handling coverage

#### 🛠️ Developer Experience
- **CLI hot-reload fix**: Replaced `tsx watch` with `nodemon` to eliminate stdin race conditions
- Separate `npm run dev:cli` and `npm run dev:cli:watch` scripts for flexibility
- Improved documentation with WSL configuration examples
- CLI smoke test examples with paste-ready JSON-RPC frames

### Technical Implementation

**New utilities:**
- `wsl-detection.ts`: Multi-signal WSL environment detection
- `host-ip-resolver.ts`: Windows host IP resolution with caching
- `connection-strategy.ts`: Network connection strategy selection logic

**Godot plugin enhancements:**
- Extended `websocket_server.gd` with bind address parameter and connection tracking
- Enhanced `plugin.gd` with UI setup, config persistence, and restart logic
- Redesigned `status_panel.tscn` with settings grid and apply button

### Additional Changes

- 📦 Dependency updates: `@modelcontextprotocol/sdk`, `hono`, `@hono/node-server`, `qs`
- 🧹 Cleanup: Removed tracked `.uid` files, improved `.gitignore` patterns
- 📝 Documentation: Updated README and CHANGELOG with WSL setup instructions and troubleshooting

### Breaking Changes

None. All changes are backward compatible with existing configurations.

### Environment Variables

- `GODOT_HOST`: Override target host address (e.g., for WSL scenarios)
- `GODOT_PORT`: Override Godot addon listen port (default: `6550`)

### CODE REVIEW

File: server/src/utils/gateway-resolver.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+/**
+ * Validates if a string is a valid IPv4 address.
+ */
+function isValidIPv4(ip: string): boolean {
+  const parts = ip.split('.');
+  if (parts.length !== 4) return false;
+  return parts.every((part) => {
+    const num = parseInt(part, 10);
+    return !Number.isNaN(num) && num >= 0 && num <= 255;
+  });
```
[the remark this answers is not in the document]
Comment (satelliteoflove) [reply server/src/utils/gateway-resolver.ts]: This is totally nit-picky; you can ignore it.
